scipy/fft/_basic_backend.py

```diff
@@ -32,6 +35,8 @@ def _execute_1D(func_str, pocketfft_func, x, n, axis, norm, overwrite_x, workers
     norm = _validate_fft_args(workers, plan, norm)
     if hasattr(xp, 'fft'):
         xp_func = getattr(xp.fft, func_str)
+        if func_str in complex_funcs:
+            x = xp_float_to_complex(x, xp)
```

Comment:
This `if` clause forces float->complex conversion on all non-numpy backends:
NumPy arrays with or without the env variables are excluded by the `if is_numpy(xp)` call above.
 
In fact, all actual backends (cupy, jax.numpy, pytorch) accept real-valued inputs, and only `array_api_strict` needs an explicit float->complex copy. So this clause forces an unneeded copy for everybody else. This should be avoided IMO, so please guard the float->complex conversion either into a try-except or an explicit `if_array_api_strict` or some such.

Reply:
a try-except sounds reasonable to me! Will re-work